### DISCUSSION TITLE
Special kind of lock that needs an open and a close.

### DISCUSSION THREAD

**Discussion by Maliron** [omersiar/esp-rfid#651] [General]
I have a special kind of lock that I need to drive a pin to open it for 150ms, and then I need to drive a different pin to close it for 150ms after the "open" time. Is there any way I can do this without hacking the code up?

Essentially, it's driving a motor forward to unlock it, leaves it open for a set time, then drives the motor in reverse to lock it. I'm using a DRV8833 to drive the motor, but I need to set 2 separate pins high and low to drive it the way I need.

**Comment by Maliron** [comment]
Guess it will have to be the hacks. I suppose I could fork it and try to add the functionality, but I'd have to learn a lot about the code. I don't think this one off use case would benefit enough people for me to do that though.

**Reply by matjack1** [reply]
I can have a look, even just to point you where you can hack something in the code to get that done, it shouldn't be too bad :)
